activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -28,11 +57,12 @@ def initialize(files, dirs = {}, &block)
             raise LoadError, "Could not load the 'listen' gem. Add `gem 'listen'` to the development group of your Gemfile", e.backtrace
           end
         end
-        Listen.to(*dtw, &method(:changed)).start
       end
+      boot!
     end
 
     def updated?
+      boot! unless @pid_hash[Process.pid]
```

Comment:
> `Concurrent::Hash` doesn't have a `compute_if_absent` locking method

The method is from `Concurrent::Map` (`Conc::Map` is a `Hash`-like data structure with a few concurrency methods thrown in, `Concurrecy::Hash` is just a synchronized carbon copy of Ruby `Hash`).

Do we even need hash for this? Can't we just have `@listen_started_on_pid` and check `Process.pid == @listen_started_on_pid`, with `@listen_started_on_pid = Process.pid` in `boot!`?